modules/tinymce/src/plugins/table/main/ts/api/Commands.ts

```diff
@@ -69,6 +69,30 @@ const registerCommands = (editor: Editor, actions: TableActions, cellSelection:
 
   const getTableFromCell = (cell: SugarElement<HTMLTableCellElement>) => TableLookup.table(cell, isRoot);
 
+  const toggleTableClass = (_ui: boolean, requestedClass: string) => {
+    getSelectionStartCellOrCaption(editor).each((startCell) => {
+      TableLookup.table(startCell, isRoot).filter(Fun.not(isRoot)).each((table) => {
+        Class.toggle(table, requestedClass);
+
+        Events.fireTableModified(editor, table.dom, Events.styleModified);
+      });
+    });
+  };
+
+  const toggleTableCellClass = (_ui: boolean, requestedClass: string) => {
+    getSelectionStartCell(editor).each((startCell) => {
+      TableLookup.table(startCell, isRoot).each((table) => {
+        const cells = TableSelection.getCellsFromSelection(startCell, selections);
+
+        Arr.each(cells, (value) => {
+          Class.toggle(value, requestedClass);
+        });
+
+        Events.fireTableModified(editor, table.dom, Events.styleModified);
+      });
+    });
+  };
```

Comment:
There's a reasonable amount of duplication here so we can simplify this (we also don't need the `Fun.not(isRoot)` from what I can see). eg:
```suggestion
  const toggleClass = (clazz: string, lookup: (table: SugarElement<HTMLTableElement>, startCell: SugarElement<HTMLTableCellElement>) => SugarElement<HTMLElement>[]) => {
    getSelectionStartCell(editor).each((startCell) => {
      TableLookup.table(startCell, isRoot).each((table) => {
        const elements = lookup(table, startCell);

        Arr.each(elements, (value) => {
          Class.toggle(value, clazz);
        });

        Events.fireTableModified(editor, table.dom, Events.styleModified);
      });
    });
  };

  const toggleTableClass = (_ui: boolean, clazz: string) =>
    toggleClass(clazz, (table, _) => [ table ]);

  const toggleTableCellClass = (_ui: boolean, clazz: string) =>
    toggleClass(clazz, (_, startCell) => TableSelection.getCellsFromSelection(startCell, selections, isRoot));
```

Note: You'll also see I added the missing `isRoot` argument to `getCellsFromSelection`.

Reply:
Also not entirely related to your changes, but we are getting this pattern duplicated a lot. As such, we should probably look at pulling that duplication out as well. eg:
```
const performActionOnSelection = (f: (table: SugarElement<HTMLTableElement>, startCell: SugarElement<HTMLTableCellElement>) => void) =>
  getSelectionStartCell(editor).each((cell) => {
    getTableFromCell(cell).each((table) => {
      f(table, cell);
    });
  });
```